### PR TITLE
Settings page

### PR DESCRIPTION
**Pull request by alonpeled44**

### CODE REVIEW

File: pages/Settings.jsx [hunk 1]
```diff
@@ -0,0 +1,89 @@
+import { useEffect, useState } from "react";
+import styles from "@/styles/pages/settings.module.scss";
+
+export default function Settings() {
+	const [theme, setTheme] = useState("default");
+	const [fontSize, setFontSize] = useState(-1);
+
+	useEffect(() => {
+		if (localStorage.getItem("theme")) {
+			if (theme === "default") {
```
Comment (liogiladi) [inline pages/Settings.jsx:10]: A flicker that happens when the page loads (both in theme and font size) because of the use of the default value - the component first renders on the default value and only updates the correct values on the 2nd/3rd rerenders (caused by the effect).
As far as I can see, there is no solution that solves this flicker that doesn't involves the server, so we'll leave the flicker for now.

File: styles/pages/settings.module.scss [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+		align-items: center;
+		gap: 10px;
+
+		h2 {
+			font-weight: 400;
+			font-size: 1.65rem;
+			color: var(--secondary-color);
+		}
+
+		.theme-switch {
```
Comment (liogiladi) [inline styles/pages/settings.module.scss:19]: relates to what I said earlier but this should move to a separate module

File: styles/pages/settings.module.scss [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+			background-color: var(--secondary-color);
+			color: var(--text-color);
+			border: none;
+			border-radius: 8px;
+			font-size: 1.35rem;
+			font-weight: 700;
+			cursor: pointer;
+		}
+
+		.font-size-switch {
```
Comment (liogiladi) [inline styles/pages/settings.module.scss:31]: relates to what I said earlier but this should move to a separate module

**Review by liogiladi — CHANGES_REQUESTED** [review]

File: pages/Settings.jsx
[the remark this answers is not in the document]
Comment (alonpeled44) [reply pages/Settings.jsx:1]: noted

File: pages/Settings.jsx [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+			return parseInt(prev);
+		});
+	};
+
+	return (
+		<main className={styles.main}>
+			<article className={styles.theme}>
+				<h2>theme -</h2>
+				<button className={styles["theme-switch"]} onClick={toggleTheme}>
+					{theme.toUpperCase()} THEME
```
[the remark this answers is not in the document]
Comment (alonpeled44) [reply pages/Settings.jsx]: noted

File: pages/Settings.jsx [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+			if (parseInt(prev) > 10) {
+				return parseInt(prev) - 1;
+			}
+			return parseInt(prev);
+		});
+	};
+
+	return (
+		<main className={styles.main}>
+			<article className={styles.theme}>
```
[the remark this answers is not in the document]
Comment (alonpeled44) [reply pages/Settings.jsx]: noted

File: pages/Settings.jsx [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+				<h2>theme -</h2>
+				<button className={styles["theme-switch"]} onClick={toggleTheme}>
+					{theme.toUpperCase()} THEME
+				</button>
+			</article>
+
+			<article>
+				<h2>font size -</h2>
+				<div className={styles["font-size-switch"]}>
```
[the remark this answers is not in the document]
Comment (alonpeled44) [reply pages/Settings.jsx]: noted

File: pages/Settings.jsx [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+			}
+			return parseInt(prev);
+		});
+	};
+
+	return (
+		<main className={styles.main}>
+			<article className={styles.theme}>
+				<h2>theme -</h2>
+				<button className={styles["theme-switch"]} onClick={toggleTheme}>
```
[the remark this answers is not in the document]
Comment (alonpeled44) [reply pages/Settings.jsx]: noted

File: pages/Settings.jsx [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+					{theme.toUpperCase()} THEME
+				</button>
+			</article>
+
+			<article>
+				<h2>font size -</h2>
+				<div className={styles["font-size-switch"]}>
+					<button onClick={handleFontPlusCLicked}>+</button>
+					<h3>{fontSize}</h3>
```
[the remark this answers is not in the document]
Comment (alonpeled44) [reply pages/Settings.jsx]: noted

File: pages/Settings.jsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (alonpeled44) [reply pages/Settings.jsx]: noted

File: pages/Settings.jsx [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+		setFontSize((prev) => {
+			if (parseInt(prev) > 10) {
+				return parseInt(prev) - 1;
+			}
+			return parseInt(prev);
+		});
+	};
+
+	return (
+		<main className={styles.main}>
```
[the remark this answers is not in the document]
Comment (alonpeled44) [reply pages/Settings.jsx]: noted

File: styles/pages/settings.module.scss (hunk 2 above)
Comment (alonpeled44) [reply styles/pages/settings.module.scss:19]: noted

File: styles/pages/settings.module.scss (hunk 3 above)
Comment (alonpeled44) [reply styles/pages/settings.module.scss:32]: noted

File: pages/Settings.jsx (hunk 6 above)
[the remark this answers is not in the document]
Comment (alonpeled44) [reply pages/Settings.jsx]: noted

File: pages/Settings.jsx (hunk 7 above)
[the remark this answers is not in the document]
Comment (alonpeled44) [reply pages/Settings.jsx]: noted

File: pages/Settings.jsx [hunk 10]
```diff
@@ -0,0 +1,89 @@
...
+		setTheme((prev) => {
+			if (prev === "bright") {
+				return changeTheme("dark");
+			} else {
+				return changeTheme("bright");
+			}
+		});
+	};
+
+	const changeTheme = (givenTheme) => {
```
[the remark this answers is not in the document]
Comment (alonpeled44) [reply pages/Settings.jsx]: noted

File: pages/Settings.jsx [hunk 11]
```diff
@@ -0,0 +1,89 @@
...
+		document.documentElement.style.setProperty("--primary-color", givenTheme === "bright" ? "#EC7373" : "#B95151");
+		document.documentElement.style.setProperty("--secondary-color", givenTheme === "bright" ? "#FFFFFF" : "#301E1E");
+		document.documentElement.style.setProperty("--text-color", givenTheme === "bright" ? "#000000" : "#FFFFFF");
+		document.documentElement.style.setProperty("--invert", givenTheme == "bright" ? "0%" : "100%");
+		return givenTheme;
+	};
+
+	const handleFontPlusCLicked = () => {
+		setFontSize((prev) => {
+			if (parseInt(prev) < 20) {
```
[the remark this answers is not in the document]
Comment (alonpeled44) [reply pages/Settings.jsx]: noted

[the remark this answers is not in the document]
Comment (alonpeled44) [reply pages/Settings.jsx]: noted

File: pages/Settings.jsx [hunk 12]
```diff
@@ -0,0 +1,89 @@
...
+
+	const changeTheme = (givenTheme) => {
+		document.documentElement.style.setProperty("--primary-color", givenTheme === "bright" ? "#EC7373" : "#B95151");
+		document.documentElement.style.setProperty("--secondary-color", givenTheme === "bright" ? "#FFFFFF" : "#301E1E");
+		document.documentElement.style.setProperty("--text-color", givenTheme === "bright" ? "#000000" : "#FFFFFF");
+		document.documentElement.style.setProperty("--invert", givenTheme == "bright" ? "0%" : "100%");
+		return givenTheme;
+	};
+
+	const handleFontPlusCLicked = () => {
```
[the remark this answers is not in the document]
Comment (alonpeled44) [reply pages/Settings.jsx]: noted

File: pages/Settings.jsx [hunk 13]
```diff
@@ -0,0 +1,89 @@
...
+		setFontSize((prev) => {
+			if (parseInt(prev) < 20) {
+				return parseInt(prev) + 1;
+			}
+			return parseInt(prev);
+		});
+	};
+
+	const handleFontMinusCLicked = () => {
```
[the remark this answers is not in the document]
Comment (alonpeled44) [reply pages/Settings.jsx]: noted

File: pages/Settings.jsx [hunk 14]
```diff
@@ -0,0 +1,89 @@
...
+			if (prev === "bright") {
+				return changeTheme("dark");
+			} else {
+				return changeTheme("bright");
+			}
+		});
+	};
+
+	const changeTheme = (givenTheme) => {
+		document.documentElement.style.setProperty("--primary-color", givenTheme === "bright" ? "#EC7373" : "#B95151");
```
[the remark this answers is not in the document]
Comment (alonpeled44) [reply pages/Settings.jsx]: noted

File: pages/Settings.jsx [hunk 15]
```diff
@@ -0,0 +1,89 @@
...
+			}
+		} else {
+			localStorage.setItem("font-size", 16);
+		}
+		document.documentElement.style.setProperty("--font-size", `${parseInt(fontSize)}px`);
+	}, [fontSize]);
+
+	const toggleTheme = () => {
+		setTheme((prev) => {
+			if (prev === "bright") {
```
[the remark this answers is not in the document]
Comment (alonpeled44) [reply pages/Settings.jsx]: noted

File: pages/Settings.jsx [hunk 16]
```diff
@@ -0,0 +1,89 @@
...
+			if (parseInt(prev) < 20) {
+				return parseInt(prev) + 1;
+			}
+			return parseInt(prev);
+		});
+	};
+
+	const handleFontMinusCLicked = () => {
+		setFontSize((prev) => {
+			if (parseInt(prev) > 10) {
```
[the remark this answers is not in the document]
Comment (alonpeled44) [reply pages/Settings.jsx]: noted

File: pages/Settings.jsx [hunk 17]
```diff
@@ -0,0 +1,89 @@
...
+		setFontSize((prev) => {
+			if (parseInt(prev) < 20) {
+				return parseInt(prev) + 1;
+			}
+			return parseInt(prev);
+		});
+	};
+
+	const handleFontMinusCLicked = () => {
+		setFontSize((prev) => {
```
[the remark this answers is not in the document]
Comment (alonpeled44) [reply pages/Settings.jsx]: noted

File: pages/Settings.jsx [hunk 18]
```diff
@@ -0,0 +1,89 @@
...
+
+	useEffect(() => {
+		if (localStorage.getItem("font-size")) {
+			if (fontSize === -1) {
+				setFontSize(localStorage.getItem("font-size"));
+			} else {
+				localStorage.setItem("font-size", fontSize);
+			}
+		} else {
+			localStorage.setItem("font-size", 16);
```
[the remark this answers is not in the document]
Comment (alonpeled44) [reply pages/Settings.jsx]: noted

File: pages/Settings.jsx (hunk 10 above)
[the remark this answers is not in the document]
Comment (alonpeled44) [reply pages/Settings.jsx]: noted

File: pages/_app.jsx [hunk 19]
```diff
@@ -1,7 +1,37 @@
...
 
 export default function App({ Component, pageProps }) {
+	useEffect(() => {
+		if (localStorage.getItem("font-size")) {
+			document.documentElement.style.setProperty("--font-size", `${parseInt(localStorage.getItem("font-size"))}px`);
+		} else {
+			document.documentElement.style.setProperty("--font-size", `${parseInt(16)}px`);
+		}
+
+		if (localStorage.getItem("theme")) {
```
[the remark this answers is not in the document]
Comment (alonpeled44) [reply pages/_app.jsx:13]: noted

File: pages/_app.jsx [hunk 20]
```diff
@@ -1,7 +1,37 @@
+import { useEffect } from "react";
 import "@/styles/globals.scss";
 import Header from "@/components/Header";
 
 export default function App({ Component, pageProps }) {
+	useEffect(() => {
+		if (localStorage.getItem("font-size")) {
+			document.documentElement.style.setProperty("--font-size", `${parseInt(localStorage.getItem("font-size"))}px`);
```
[the remark this answers is not in the document]
Comment (alonpeled44) [reply pages/_app.jsx]: noted

File: pages/Settings.jsx [hunk 21]
```diff
@@ -0,0 +1,89 @@
...
+			} else {
+				localStorage.setItem("theme", theme);
+			}
+		} else {
+			localStorage.setItem("theme", "bright");
+		}
+		changeTheme(theme);
+	}, [theme]);
+
+	useEffect(() => {
```
[the remark this answers is not in the document]
Comment (alonpeled44) [reply pages/Settings.jsx]: noted

File: pages/_app.jsx [hunk 22]
```diff
@@ -1,7 +1,37 @@
+import { useEffect } from "react";
 import "@/styles/globals.scss";
 import Header from "@/components/Header";
 
 export default function App({ Component, pageProps }) {
+	useEffect(() => {
```
[the remark this answers is not in the document]
Comment (alonpeled44) [reply pages/_app.jsx:18]: noted

File: pages/Settings.jsx [hunk 23]
```diff
@@ -0,0 +1,89 @@
+import { useEffect, useState } from "react";
+import styles from "@/styles/pages/settings.module.scss";
+
+export default function Settings() {
+	const [theme, setTheme] = useState("default");
+	const [fontSize, setFontSize] = useState(-1);
+
+	useEffect(() => {
```
[the remark this answers is not in the document]
Comment (alonpeled44) [reply pages/Settings.jsx]: noted

File: pages/Settings.jsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (alonpeled44) [reply pages/Settings.jsx]: noted

Comment (alonpeled44) [reply pages/Settings.jsx]: noted

File: components/Changers/ThemeChanger.jsx [hunk 24]
```diff
@@ -0,0 +1,21 @@
+import styles from "@/styles/components/changers/theme-changer.module.scss";
+
+export default function ThemeChanger({ theme, setTheme }) {
+	const updateThemeVariables = () => {
+		setTheme((prev) => {
+			if (prev === "bright") {
```
Comment (liogiladi) [inline components/Changers/ThemeChanger.jsx:6]: Suggests this:

```js
setTheme((prev) => {
    const newValue = prev === "bright" ? "dark" : "bright";
    localStorage["theme"] = newValue;
    return newValue;
});
```

File: components/Setter.jsx [hunk 25]
```diff
@@ -0,0 +1,10 @@
+import styles from "@/styles/components/setter.module.scss";
+
+export default function Setter({ header, changer }) {
```
Comment (liogiladi) [inline components/Setter.jsx:3]: This component is not a good idea.
1. It misleads the actual purpose of the component - not to set something but to layout (UI wise) the "setter" in a certain way.
2. It obfuscates simple the behaviour and structure of the layout (It will be more coherent to look at the settings page and see directly the structure).
3. What is the purpose of `header`? Is it a header? not really, it's just the name of the setter.
4. `changerElement` would be a better name to indicate  that it is a element/component you are passing, but regardless it's not a good idea to pass the setter that way (combined with the whole purpose of the component).

The bottom line is that not all behaviour that is repeating should be split/encapsulated.

File: pages/_app.jsx [hunk 26]
```diff
@@ -1,7 +1,50 @@
+import { useEffect, useState } from "react";
 import "@/styles/globals.scss";
 import Header from "@/components/Header";
 
 export default function App({ Component, pageProps }) {
+	const [theme, setTheme] = useState("bright");
+	const [fontSize, setFontSize] = useState(16);
```
Comment (liogiladi) [inline pages/_app.jsx:7]: mentioned this in another comment but you shouldn't use a state here for the font size as you only use the local storage or the default value.

This is apparent if you look at the current chain of event:
first render: set font state ---effect---> second render: set font css variable 
there's no need for 2 steps as you can just do the second one

File: pages/_app.jsx [hunk 27]
```diff
@@ -1,7 +1,50 @@
+import { useEffect, useState } from "react";
 import "@/styles/globals.scss";
 import Header from "@/components/Header";
 
 export default function App({ Component, pageProps }) {
+	const [theme, setTheme] = useState("bright");
```
Comment (liogiladi) [inline pages/_app.jsx:6]: The idea of the state residing in the app.jsx is correct but it should be inside a custom context to enable the global access that prevent prop drilling.

File: pages/_app.jsx [hunk 28]
```diff
@@ -1,7 +1,50 @@
...
+	const [fontSize, setFontSize] = useState(16);
+
+	const brightPrimary = "#EC7373";
```
Comment (liogiladi) [inline pages/_app.jsx:9]: these variables don't belong to this file. mentioned this before but you should separate them to a different file or colocate them inside the themeContext

File: pages/_app.jsx [hunk 29]
```diff
@@ -1,7 +1,50 @@
...
+
+	useEffect(() => {
+		if (localStorage.getItem("font-size")) {
+			setFontSize(localStorage.getItem("font-size"));
+		} else {
+			localStorage.setItem("font-size", 16);
+		}
+
+		if (localStorage.getItem("theme")) {
+			setTheme(localStorage.getItem("theme"));
```
Comment (liogiladi) [inline pages/_app.jsx:26]: don't know if it will be relavent after refactoring but you can also do:
```js
setTheme(localStorage.getItem("theme") || "bright");
```
instead of the condional, the upside, besides looks, is that here you access the localStorage only once

File: pages/_app.jsx [hunk 30]
```diff
@@ -1,7 +1,50 @@
...
+		} else {
+			setTheme("bright");
+		}
+	}, []);
+
+	useEffect(() => {
```
Comment (liogiladi) [inline pages/_app.jsx:32]: should be inside the context

File: pages/_app.jsx [hunk 31]
```diff
@@ -1,7 +1,50 @@
...
+		document.documentElement.style.setProperty("--secondary-color", theme === "bright" ? brightSecondary : darkSecondary);
+		document.documentElement.style.setProperty("--text-color", theme === "bright" ? brightText : darkText);
+		document.documentElement.style.setProperty("--invert", theme === "bright" ? brightInvert : darkInvert);
+	}, [theme]);
+
+	useEffect(() => {
+		document.documentElement.style.setProperty("--font-size", `${parseInt(fontSize)}px`);
+	}, [fontSize]);
+
+	pageProps["theme"] = theme;
```
Comment (liogiladi) [inline pages/_app.jsx:43]: explained in another comment but not a good idea

File: pages/settings.jsx [hunk 32]
```diff
@@ -0,0 +1,13 @@
+import styles from "@/styles/pages/settings.module.scss";
+import Setter from "@/components/Setter";
+import ThemeChanger from "@/components/Changers/themeChanger";
+import FontSizeChanger from "@/components/Changers/FontSizeChanger";
+
+export default function Settings({ fontSize, setFontSize, theme, setTheme }) {
+	return (
+		<main id={styles.main}>
+			<Setter header='theme' changer={<ThemeChanger theme={theme} setTheme={setTheme} />} />
```
Comment (liogiladi) [inline pages/settings.jsx:9]: explained in a comment inside `<Setter>` but you should just include the sections here

File: components/Changers/FontSizeChanger.jsx [hunk 33]
```diff
@@ -0,0 +1,25 @@
+import styles from "@/styles/components/changers/font-size-changer.module.scss";
+
+export default function FontSizeChanger({ fontSize, setFontSize }) {
```
Comment (liogiladi) [inline components/Changers/FontSizeChanger.jsx:3]: Don't remember if I commented on it elsewhere but a better name would be `CounterInput` or something

Comment (liogiladi) [inline components/Changers/FontSizeChanger.jsx:3]: The state should be in here (descibed about it more when talked about the context/custom hook)

File: components/Changers/FontSizeChanger.jsx
Comment (liogiladi) [inline components/Changers/FontSizeChanger.jsx:1]: after all refactoring, there is no need for the "Changes" folder as  the components  will be general.
Also notice that "Changers" is inconsistent with the casing

**Review by liogiladi — CHANGES_REQUESTED** [review]

File: pages/Settings.jsx (hunk 23 above)
[the remark this answers is not in the document]
Comment (alonpeled44) [reply pages/Settings.jsx]: noted

File: pages/Settings.jsx (hunk 7 above)
[the remark this answers is not in the document]
Comment (alonpeled44) [reply pages/Settings.jsx]: noted

File: pages/Settings.jsx (hunk 6 above)
[the remark this answers is not in the document]
Comment (alonpeled44) [reply pages/Settings.jsx]: noted

File: pages/_app.jsx (hunk 22 above)
[the remark this answers is not in the document]
Comment (alonpeled44) [reply pages/_app.jsx:11]: noted

File: pages/Settings.jsx (hunk 21 above)
[the remark this answers is not in the document]
Comment (alonpeled44) [reply pages/Settings.jsx]: noted

File: pages/_app.jsx (hunk 19 above)
[the remark this answers is not in the document]
Comment (alonpeled44) [reply pages/_app.jsx:18]: noted

File: components/Changers/FontSizeChanger.jsx (hunk 33 above)
[the remark this answers is not in the document]
Comment (alonpeled44) [reply components/Changers/FontSizeChanger.jsx]: noted

File: components/Changers/ThemeChanger.jsx [hunk 34]
```diff
@@ -0,0 +1,21 @@
+import styles from "@/styles/components/changers/theme-changer.module.scss";
+
+export default function ThemeChanger({ theme, setTheme }) {
```
[the remark this answers is not in the document]
Comment (alonpeled44) [reply components/Changers/ThemeChanger.jsx]: noted

File: components/Changers/ThemeChanger.jsx [hunk 35]
```diff
@@ -0,0 +1,21 @@
...
+	const updateThemeVariables = () => {
```
[the remark this answers is not in the document]
Comment (alonpeled44) [reply components/Changers/ThemeChanger.jsx]: noted

File: components/Changers/ThemeChanger.jsx (hunk 24 above)
Comment (alonpeled44) [reply components/Changers/ThemeChanger.jsx]: noted

File: components/Setter.jsx (hunk 25 above)
Comment (alonpeled44) [reply components/Setter.jsx]: noted

File: pages/_app.jsx (hunk 26 above)
Comment (alonpeled44) [reply pages/_app.jsx:9]: changed?

File: pages/_app.jsx (hunk 28 above)
Comment (alonpeled44) [reply pages/_app.jsx]: noted

File: pages/_app.jsx (hunk 29 above)
Comment (alonpeled44) [reply pages/_app.jsx:19]: doesnt work, says not defined

File: pages/_app.jsx (hunk 30 above)
Comment (alonpeled44) [reply pages/_app.jsx:25]: changed?

File: pages/_app.jsx (hunk 31 above)
Comment (alonpeled44) [reply pages/_app.jsx]: noted

File: pages/settings.jsx (hunk 32 above)
Comment (alonpeled44) [reply pages/settings.jsx]: noted

File: components/Changers/FontSizeChanger.jsx (hunk 33 above)
Comment (alonpeled44) [reply components/Changers/FontSizeChanger.jsx]: noted

Comment (alonpeled44) [reply components/Changers/FontSizeChanger.jsx]: changed?

File: components/Changers/FontSizeChanger.jsx
Comment (alonpeled44) [reply components/Changers/FontSizeChanger.jsx:1]: noted

File: pages/_app.jsx (hunk 26 above)
Comment (liogiladi) [reply pages/_app.jsx:9]: no? the state is still in `_app.jsx`. What was meant is that the state should be in the font size component and here you only listen get the value from local storage. In the component you change in runtime the value in css via useEffect

File: pages/_app.jsx (hunk 29 above)
Comment (liogiladi) [reply pages/_app.jsx:19]: It does work, which make sense:

```js
if(value) {
     set(value);
} else {
     set("default");
}
```

is equivalent to this:
```js
set(value || "default");
``` 

the `||` returns the second value if the first one is falsy

File: pages/_app.jsx [hunk 36]
```diff
@@ -1,11 +1,45 @@
...
+		document.documentElement.style.setProperty(
+			"--secondary-color",
+			theme === "bright" ? colors.brightSecondary : colors.darkSecondary
+		);
+		document.documentElement.style.setProperty("--text-color", theme === "bright" ? colors.brightText : colors.darkText);
+		document.documentElement.style.setProperty("--invert", theme === "bright" ? colors.brightInvert : colors.darkInvert);
+	}, [theme]);
+
+	useEffect(() => {
+		document.documentElement.style.setProperty("--font-size", `${parseInt(fontSize)}px`);
```
Comment (liogiladi) [inline pages/_app.jsx:36]: there's no need to parse the fontSize from localStorage as you use it as string one way or another

File: pages/_app.jsx [hunk 37]
```diff
@@ -1,11 +1,45 @@
...
+	}, [fontSize]);
+
 	return (
-		<>
+		<PageFeatures.Provider value={{ theme: theme, setTheme: setTheme, fontSize: fontSize, setFontSize: setFontSize }}>
```
Comment (liogiladi) [inline pages/_app.jsx:40]: this:
```js
value={{ theme: theme, setTheme: setTheme, fontSize: fontSize, setFontSize: setFontSize }}
```

can be simplified to this:
```js
value={{ theme, setTheme, fontSize, setFontSize }}
```

File: components/ThemeToggleButton.jsx [hunk 38]
```diff
@@ -0,0 +1,22 @@
+import { useContext } from "react";
+import styles from "@/styles/components/theme-toggle-button.module.scss";
+import PageFeatures from "@/server/context";
+import ToggleButton from "@/components/ToggleButton";
+
+export default function ThemeToggleButton() {
+	const theme = useContext(PageFeatures)["theme"];
+	const setTheme = useContext(PageFeatures)["setTheme"];
+	const ToggleTheme = () => {
```
Comment (liogiladi) [inline components/ThemeToggleButton.jsx:9]: pedantic warning! pls put whitespace between above and below this function :')

File: components/ThemeToggleButton.jsx [hunk 39]
```diff
@@ -0,0 +1,22 @@
...
+		setTheme((prev) => {
+			const newValue = prev === "bright" ? "dark" : "bright";
+			localStorage.setItem("theme", newValue);
+			return newValue;
+		});
+	};
+	return (
+		<section className={styles["toggle-theme"]}>
```
Comment (liogiladi) [inline components/ThemeToggleButton.jsx:17]: as I described in another comment, the section, h2 should be in the page itself.
this component is only a wrapper to ToggleButton

File: pages/settings.jsx (hunk 32 above)
Comment (liogiladi) [reply pages/settings.jsx]: didn't do that aye? >:)

File: pages/Settings.jsx (hunk 21 above)
[the remark this answers is not in the document]
Comment (liogiladi) [reply pages/Settings.jsx]: didn't do that (described this in a newer comment aswell)

File: server/colors.js
Comment (liogiladi) [inline server/colors.js:1]: why is the file in server folder? should be at root or a new folder like `utils`

File: server/colors.js [hunk 40]
```diff
@@ -0,0 +1,12 @@
+const colors = {
```
Comment (liogiladi) [inline server/colors.js:1]: The way you implement it is good, so you don't need to change it.

Just want to show you a different way that I think is preferable:
```js
const colors = {
   bright: {
      primary: "#EC7373",
      ...
   },
   dark: {
      primary: "#B95151",
      ...
   }
};
```

this way instead of doing this (pseudo code):
```js
setColor(theme == "bright" ? colors.brightPrimary : colors.darkPrimary);
```

you can just do this:
```js
setColor(colors[theme].primary);
```

which, in my opinion, is cleaner and clearer.
You can resolve this conversation right away.

File: server/context.jsx [hunk 41]
```diff
@@ -0,0 +1,5 @@
+import { createContext } from "react";
+
+const PageFeatures = createContext(null);
```
Comment (liogiladi) [inline server/context.jsx:3]: There is no need to separate the context from `_app.jsx` if you don't create a custom hook out of it or use more complex logic.
This should be in `_app.jsx` at the top.

File: components/FontSizeCounterInput.jsx [hunk 42]
```diff
@@ -0,0 +1,37 @@
+import { useContext } from "react";
+import styles from "@/styles/components/font-size-counter-input.module.scss";
+import PageFeatures from "@/server/context";
+import CounterInput from "@/components/CounterInput";
+export default function FontSizeCounterInput() {
```
Comment (liogiladi) [inline components/FontSizeCounterInput.jsx:5]: pedantic warning! \n? UwU

File: server/context.jsx (hunk 41 above)
Comment (liogiladi) [inline server/context.jsx:3]: Note for future reference that you put the context in the server folder - it is a client thing only so it doesn't belong there. There will be probably a folder named `contexts` for those

**Review by liogiladi — CHANGES_REQUESTED** [review]

File: pages/_app.jsx (hunk 26 above)
Comment (alonpeled44) [reply pages/_app.jsx]: noted

File: pages/_app.jsx (hunk 29 above)
Comment (alonpeled44) [reply pages/_app.jsx]: noted, different usage

File: pages/settings.jsx (hunk 32 above)
Comment (alonpeled44) [reply pages/settings.jsx]: noted

File: components/CounterInput.jsx [hunk 43]
```diff
@@ -0,0 +1,11 @@
+import styles from "@/styles/components/counter-input.module.scss";
+
+export default function CounterInput({ onIncrease, onDecrease, children }) {
+	return (
+		<div className={styles["input-counter"]}>
+			<button onClick={onIncrease}>+</button>
+			<p>{children}</p>
```
[the remark this answers is not in the document]
Comment (alonpeled44) [reply components/CounterInput.jsx]: noted

File: components/FontSizeCounterInput.jsx [hunk 44]
```diff
@@ -0,0 +1,37 @@
...
+	const fontSize = parseInt(useContext(PageFeatures)["fontSize"]);
```
[the remark this answers is not in the document]
Comment (alonpeled44) [reply components/FontSizeCounterInput.jsx]: noted

File: components/FontSizeCounterInput.jsx [hunk 45]
```diff
@@ -0,0 +1,37 @@
...
+	const setFontSize = useContext(PageFeatures)["setFontSize"];
+
+	const increaseFontsize = () => {
+		if (fontSize < 20) {
+			setFontSize((prev) => {
+				const newFontSize = parseInt(prev) + 1;
+				localStorage.setItem("fontSize", newFontSize);
```
[the remark this answers is not in the document]
Comment (alonpeled44) [reply components/FontSizeCounterInput.jsx:19]: noted

File: components/FontSizeCounterInput.jsx [hunk 46]
```diff
@@ -0,0 +1,37 @@
...
+				return newFontSize;
+			});
+		}
+	};
+
+	return (
+		<section className={styles["font-size-counter"]}>
+			<h2>font size - </h2>
+			<CounterInput onIncrease={increaseFontsize} onDecrease={decreaseFontSize}>
```
[the remark this answers is not in the document]
Comment (alonpeled44) [reply components/FontSizeCounterInput.jsx]: noted

[the remark this answers is not in the document]
Comment (alonpeled44) [reply components/FontSizeCounterInput.jsx]: noted

File: components/ThemeToggleButton.jsx [hunk 47]
```diff
@@ -0,0 +1,22 @@
+import { useContext } from "react";
+import styles from "@/styles/components/theme-toggle-button.module.scss";
+import PageFeatures from "@/server/context";
+import ToggleButton from "@/components/ToggleButton";
+
+export default function ThemeToggleButton() {
+	const theme = useContext(PageFeatures)["theme"];
```
[the remark this answers is not in the document]
Comment (alonpeled44) [reply components/ThemeToggleButton.jsx]: noted

File: pages/_app.jsx (hunk 37 above)
Comment (alonpeled44) [reply pages/_app.jsx]: noted

File: components/ThemeToggleButton.jsx (hunk 38 above)
Comment (alonpeled44) [reply components/ThemeToggleButton.jsx:8]: noted

File: components/ThemeToggleButton.jsx (hunk 39 above)
Comment (alonpeled44) [reply components/ThemeToggleButton.jsx]: noted

File: server/colors.js
Comment (alonpeled44) [reply server/colors.js:1]: noted

File: server/colors.js (hunk 40 above)
Comment (alonpeled44) [reply server/colors.js]: this is way better if there were more themes, noted

File: server/context.jsx (hunk 41 above)
Comment (alonpeled44) [reply server/context.jsx]: noted

File: components/FontSizeCounterInput.jsx (hunk 42 above)
Comment (alonpeled44) [reply components/FontSizeCounterInput.jsx:4]: I know you start to enjoy finding those 🙌

File: server/context.jsx (hunk 41 above)
Comment (alonpeled44) [reply server/context.jsx]: not relevant anymore, but noted

File: components/FontSizeCounterInput.jsx [hunk 48]
```diff
@@ -0,0 +1,36 @@
+import { useEffect, useState } from "react";
+import CounterInput from "@/components/CounterInput";
+
+export default function FontSizeCounterInput() {
+	const [fontSize, setFontSize] = useState(16);
+
+	useEffect(() => {
```
Comment (liogiladi) [inline components/FontSizeCounterInput.jsx:7]: You can remove this effect and it won't change a thing. This is because the effect below runs with the initial value of the state - 16

File: components/FontSizeCounterInput.jsx
Comment (liogiladi) [inline components/FontSizeCounterInput.jsx:1]: Would be better to put this and the other variant component in a separate folder inside `components` folder named `settings`.
This is preferable because you don't mix that way global components with specific use ones

File: components/FontSizeCounterInput.jsx [hunk 49]
```diff
@@ -0,0 +1,36 @@
...
+				localStorage.setItem("fontSize", newFontSize);
+				return newFontSize;
+			});
+		}
+	};
+
+	const decreaseFontSize = () => {
+		if (fontSize > 10) {
+			setFontSize((prev) => {
+				const newFontSize = parseInt(prev) - 1;
```
Comment (liogiladi) [inline components/FontSizeCounterInput.jsx:28]: The same here with the parsing

**Review by liogiladi — CHANGES_REQUESTED** [review]

File: components/FontSizeCounterInput.jsx (hunk 48 above)
Comment (alonpeled44) [reply components/FontSizeCounterInput.jsx:7]: not the reason, is needed

File: components/FontSizeCounterInput.jsx [hunk 50]
```diff
@@ -0,0 +1,37 @@
...
+import PageFeatures from "@/server/context";
+import CounterInput from "@/components/CounterInput";
+export default function FontSizeCounterInput() {
+	const fontSize = parseInt(useContext(PageFeatures)["fontSize"]);
+	const setFontSize = useContext(PageFeatures)["setFontSize"];
+
+	const increaseFontsize = () => {
+		if (fontSize < 20) {
+			setFontSize((prev) => {
+				const newFontSize = parseInt(prev) + 1;
```
[the remark this answers is not in the document]
Comment (alonpeled44) [reply components/FontSizeCounterInput.jsx:18]: noted

File: components/FontSizeCounterInput.jsx
Comment (alonpeled44) [reply components/FontSizeCounterInput.jsx:1]: noted

**Review by liogiladi — APPROVED** [review]
yay
